airbyte-server/src/main/java/io/airbyte/server/handlers/MigrationHandler.java

```diff
@@ -0,0 +1,181 @@
+    this.configRepository = configRepository;
+  }
+
+  public Path exportData() {
+    try {
+      final Path tempFolder = Files.createTempDirectory("airbyte_archive");
+      final Path archive = Files.createTempFile(ARCHIVE_FILE_NAME, ".tar.gz");;
+      archive.toFile().deleteOnExit();
+      try {
+        exportVersionFile(tempFolder);
+        exportAirbyteConfig(tempFolder);
+        exportAirbyteDatabase(tempFolder);
+        ArchiveHelper.createArchive(tempFolder, archive);
+      } finally {
+        FileUtils.deleteDirectory(tempFolder.toFile());
+      }
+      return archive;
+    } catch (IOException e) {
+      LOGGER.error("Export Data failed.");
+      throw new RuntimeException(e);
+    }
+  }
+
+  private void exportVersionFile(Path tempFolder) throws IOException {
+    final File versionFile = Files.createFile(tempFolder.resolve(VERSION_FILE_NAME)).toFile();
+    FileUtils.writeStringToFile(versionFile, version, Charset.defaultCharset());
+  }
+
+  private void exportAirbyteConfig(Path tempFolder) {
+    LOGGER.info("Exporting Airbyte Configs");
+    final ConfigConverter configConverter = new ConfigConverter(version, tempFolder);
+    Exceptions.toRuntime(() -> {
+      configConverter.writeConfigList(ConfigSchema.STANDARD_WORKSPACE,
+          List.of(configRepository.getStandardWorkspace(PersistenceConstants.DEFAULT_WORKSPACE_ID)));
+      configConverter.writeConfigList(ConfigSchema.STANDARD_SOURCE_DEFINITION, configRepository.listStandardSources());
+      configConverter.writeConfigList(ConfigSchema.STANDARD_DESTINATION_DEFINITION, configRepository.listStandardDestinationDefinitions());
+      configConverter.writeConfigList(ConfigSchema.SOURCE_CONNECTION, configRepository.listSourceConnection());
+      configConverter.writeConfigList(ConfigSchema.DESTINATION_CONNECTION, configRepository.listDestinationConnection());
+      final List<StandardSync> standardSyncs = configRepository.listStandardSyncs();
+      configConverter.writeConfigList(ConfigSchema.STANDARD_SYNC, standardSyncs);
+      final List<StandardSyncSchedule> standardSchedules = standardSyncs
+          .stream()
+          .map(config -> Exceptions.toRuntime(() -> configRepository.getStandardSyncSchedule(config.getConnectionId())))
+          .filter(Objects::nonNull)
+          .collect(Collectors.toList());
+      configConverter.writeConfigList(ConfigSchema.STANDARD_SYNC_SCHEDULE, standardSchedules);
+    });
+  }
+
+  private void exportAirbyteDatabase(Path tempFolder) {
+    LOGGER.info("Exporting Airbyte Database");
+    // TODO implement
+  }
+
+  public void importData(Path archive) {
+    try {
+      final Path tempFolder = Files.createTempDirectory("airbyte_archive");
+      try {
+        ArchiveHelper.openArchive(tempFolder, archive);
+        checkImport(tempFolder);
+        importAirbyteConfig(tempFolder, false);
+        importAirbyteDatabase(tempFolder, false);
+      } finally {
+        FileUtils.deleteDirectory(tempFolder.toFile());
```

Comment:
should we delete the archive too?

Reply:
Yes we can and i guess you find a good way to do so here: https://github.com/airbytehq/airbyte/pull/1591

Reply:
i think that's a separate issue. I think for import once we have processed the data we can just delete the file. we don't have to do all of this ttl stuff. i think it would be okay if you just add `Files.delete(archive)` (or whatever the syntax is.